### PR TITLE
Apply Multi-GPU/TPU Normalization

### PR DESCRIPTION
**Pull request by MokkeMeguru**
In Tensorflow 2.2.0, they implement multi-gpu/tpu's batch normalization layer, So I implement Weight Normalization Layer by reference to it.
Sources:
https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/keras/layers/normalization_v2.py#L165-L204

### CODE REVIEW

**Comment by MokkeMeguru** [comment]
Sorry, I am suffering from your Formatter...

File: tensorflow_addons/layers/wrappers.py [hunk 1]
```diff
@@ -182,13 +180,56 @@ def _init_norm(self):
             g_tensor = self.g.assign(tf.reshape(v_norm, (self.layer_depth,)))
             return [g_tensor]
 
+    def _calculate_moments(self, x, axes, keep_dims=True):
```
Comment (bhack) [inline tensorflow_addons/layers/wrappers.py:183]: Can we ask to expose this in Tensorflow directly?

Comment (MokkeMeguru) [reply tensorflow_addons/layers/wrappers.py:183]: I think Tensorflow has a too complex test system. (In my other pull request, I had a very hard time.)
And also, its framework always doesn't have backward compatibility in particular with Multi-GPU/TPU.

So I have no enough power for sending the pull request for TensorFlow...

Comment (bhack) [reply tensorflow_addons/layers/wrappers.py:183]: I don't know if you could push more effort here cause https://github.com/tensorflow/addons/pull/1789#issuecomment-625876871. /cc @gabrieldemarmiesse

Comment (gabrieldemarmiesse) [reply tensorflow_addons/layers/wrappers.py:183]: Yeah, we need to rethink the implementation of `WeightNormalization`. To the best of my knowledge it's impossible to hook ourselves because the use of a weight in a layer without using hacky stuff that will come back to bite us later.

We'll likely have to deprecate the wrapper layer and provide utility functions instead, with a good tutorial to explain how the use need to do the subclassing for it to work. We could maybe provide a new Dense layer with the weight normalization implemented inside. That would cover many use cases and provide an example of how to do it for other layers.

It would be nice if somebody could work on this and provide a proposal/possibly a new API to fix all those issues.

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
